python/tests/wrapper/test_01_synthetic_collection.py

```diff
@@ -0,0 +1,148 @@
+    config = messages.TopTokensScoreConfig()
+    config.num_tokens = num_top_tokens
+    ref_score_config.config = config.SerializeToString()
+
+    master_id = lib.ArtmCreateMasterComponent(master_config)
+
+    # Initialize model
+    init_args = messages.InitializeModelArgs()
+    init_args.model_name = pwt
+    init_args.disk_path = batches_folder
+    init_args.source_type = constants.InitializeModelArgs_SourceType_Batches
+    init_args.topics_count = num_topics
+    lib.ArtmInitializeModel(master_id, init_args)
+
+    # Create configuration for batch processing
+    proc_args = messages.ProcessBatchesArgs()
+    proc_args.pwt_source_name = pwt
+    proc_args.nwt_target_name = nwt
+
+    for name in os.listdir(batches_folder):
+        proc_args.batch_filename.append(os.path.join(batches_folder, name))
+    proc_args.inner_iterations_count = inner_iterations_count
+
+    # Create configuration for Phi normalization
+    norm_args = messages.NormalizeModelArgs()
+    norm_args.pwt_target_name = pwt
+    norm_args.nwt_source_name = nwt
+
+    # Create config for scores retrieval
+    perplexity_args = messages.GetScoreValueArgs()
+    perplexity_args.model_name = pwt
+    perplexity_args.score_name = 'PerplexityScore'
+
+    top_tokens_args = messages.GetScoreValueArgs()
+    top_tokens_args.model_name = pwt
+    top_tokens_args.score_name = 'TopTokensScore'
+
+    for iter in xrange(outer_iterations_count):
+        # Invoke one scan of the collection and normalize Phi
+        lib.ArtmRequestProcessBatches(master_id, proc_args)
+        lib.ArtmNormalizeModel(master_id, norm_args)    
+
+        # Retrieve and print perplexity score
+        results = lib.ArtmRequestScore(master_id, perplexity_args)
+        score_data = messages.ScoreData()
+        score_data.ParseFromString(results)
+        perplexity_score = messages.PerplexityScore()
+        perplexity_score.ParseFromString(score_data.data)
+
+        assert abs(perplexity_score.value - expected_perplexity_value_on_iteration[iter]) < perplexity_tol
+        print 'Iteration#{0} : Perplexity = {1:.3}'.format(str(iter), perplexity_score.value)
+
+    # Retrieve and print top tokens score
+    results = lib.ArtmRequestScore(master_id, top_tokens_args)
+    score_data = messages.ScoreData()
+    score_data.ParseFromString(results)
+    top_tokens_score = messages.TopTokensScore()
+    top_tokens_score.ParseFromString(score_data.data)
+
+    print 'Top tokens per topic:'
+    top_tokens_triplets = zip(top_tokens_score.topic_index, zip(top_tokens_score.token, top_tokens_score.weight))
+    for topic_index, group in itertools.groupby(top_tokens_triplets, key=lambda (topic_index, _): topic_index):
+        string = 'Topic#{0} : '.format(topic_index)
+        for _, (token, weight) in group:
+            assert abs(weight - expected_top_tokens_weight) < top_tokens_tol
+            string += ' {0}({1:.3})'.format(token, weight)
+        print string
+
+    shutil.rmtree(batches_folder)
```

Comment:
Hm. This is dangerous because in the case of exception in the code above your temp folder will not be removed.
Here discussed a more robust way of working with temporary dir: [stackoverflow](http://stackoverflow.com/questions/13379742/right-way-to-clean-up-a-temporary-folder-in-python-class).